<launch>

    <!-- vehicle name e.g. cc2017_car -->
    <arg name="vehicle" default="cc2017_car"/>

    <!-- topic names -->
    <arg name="camera" default="camera"/>
    <arg name="image_topic" default="/$(arg camera)/image_raw"/>
    <arg name="homo_topic" default="/$(arg camera)/camera_homography"/>
    <arg name="odometry_topic" default="/visual_odom/odom" />
    <arg name="debug_image_topic" default="/visual_odom/debug" />
    <arg name="cam_info_topic" default=""/> <!-- if empty: it will be assumed that the image is already rectified and no undistortion is being done -->


    <!-- tf parent frame of published tf -->
    <arg name="static_frame" default="odom"/>

    <!-- tf child frame of published tf -->
    <arg name="moving_frame" default="rear_axis_middle_ground"/>

    <!-- vehicle configuration file it should be named: <vehicle>.yaml -->
    <arg name="vehicle_config" default="$(find drive_ros_localize_visual_odometry)/config/$(arg vehicle).yaml" />

    <!-- forward output to [screen|log] -->
    <arg name="output" default="screen"/>

    <node name="simple_visual_odometry" pkg="drive_ros_localize_visual_odometry"
          type="simple_visual_odometry" respawn="false" output="$(arg output)">
        <remap from="~/img_in" to="$(arg image_topic)"/>
        <remap from="~/homog" to="$(arg homo_topic)"/>
        <remap from="~/odom_out" to="$(arg odometry_topic)" />
        <remap from="~/img_out" to="$(arg debug_image_topic)" />
        <param name="cam_info_topic" type="str" value="$(arg cam_info_topic)"/> <!-- handle as parameter because could be empty -->
        <param name="static_frame" type="str" value="$(arg static_frame)" />
        <param name="moving_frame" type="str" value="$(arg moving_frame)" />
        <rosparam command="load" file="$(arg vehicle_config)"/>
    </node>
</launch>
